client/snapctl.go

```diff
@@ -35,14 +48,36 @@ type SnapCtlOptions struct {
 	Args []string `json:"args"`
 }
 
+// SnapCtlPostData is the data posted to the daemon /v2/snapctl endpoint
+// TODO: this can be removed again once we no longer need to pass stdin data
+//       but instead use a real stdin stream
+type SnapCtlPostData struct {
+	SnapCtlOptions
+
+	Stdin []byte `json:"stdin,omitempty"`
+}
+
 type snapctlOutput struct {
 	Stdout string `json:"stdout"`
 	Stderr string `json:"stderr"`
 }
 
 // RunSnapctl requests a snapctl run for the given options.
-func (client *Client) RunSnapctl(options *SnapCtlOptions) (stdout, stderr []byte, err error) {
-	b, err := json.Marshal(options)
+func (client *Client) RunSnapctl(options *SnapCtlOptions, stdin io.Reader) (stdout, stderr []byte, err error) {
+	// TODO: instead of reading all of stdin here we need to forward it to
+	//       the daemon eventually
+	var stdinData []byte
+	if stdin != nil {
+		stdinData, err = ioutil.ReadAll(stdin)
```

Comment:
Should we impose a limit here in case the caller misbehaves (and more importantly, on the daemon side)?

Reply:
I will do this as a followup, I like the idea to limit this to e.g. 1Mb or something huge to avoid this being totally unbounded.